pkg/segment/query/iqr/intermediateQueryResult.go

```diff
@@ -395,6 +395,28 @@ func (iqr *IQR) Append(other *IQR) error {
 	return nil
 }
 
+func (iqr *IQR) GetColumns() (map[string]struct{}, error) {
+	if err := iqr.validate(); err != nil {
+		return nil, toputils.TeeErrorf("IQR.GetColumns: validation failed: %v", err)
+	}
+
+	segKey, ok := iqr.encodingToSegKey[iqr.rrcs[0].SegKeyInfo.SegKeyEnc]
+	if !ok {
+		return nil, toputils.TeeErrorf("IQR.readColumnWithRRCs: unknown encoding %v", iqr.rrcs[0].SegKeyInfo.SegKeyEnc)
+	}
+
+	vTable := iqr.rrcs[0].VirtualTableName
+
+	allColumns, err := record.GetColsForSegKey(segKey, vTable)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
You're making an assumption here that all the RRCs in the IQR belong to the same segment. That's not true in general. An IQR is just a list of records—those records can come from anywhere. You may want to check the `iqr.encodingToSegKey` to get all the segkeys. However, while that map has all the segkeys used by the `iqr.RRCs`, it may have some segkeys which are not used by any of the RRCs.

Another potential option would be to iterate through the `iqr.RRCs` and get all the segkeys and VTable names